### PR TITLE
dev/sg: clean up linter structure

### PR DESCRIPTION
**Pull request by bobheadxi**
Gives linters a similar structure to `checks`, moving core types and utility funcs into a separate package and moving actual lint implementations to a separate file.

Also fixes a bug where linter flags weren't being propagated correctly

## Test plan

```
sg update -local
sg lint
```

The help text has also improved by making sure `ShortUsage` is set and `ShortHelp` and `LongHelp` are both set to Help`

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #32212** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/32212" target="_blank">Graphite</a> 
    * **PR #32213** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/32213" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Review by mrnugget — APPROVED** [review]
Nice! Squeaky clean

File: dev/sg/internal/lint/lint.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import (
+	"context"
+	"flag"
+	"time"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/run"
+)
+
+// Func is a linter.
+type Func func(context.Context) *Report
```
Comment (jhchabran) [inline dev/sg/internal/lint/lint.go:12]: nit: what about calling this `Runner` to make the intention of this type clearer?

**Review by jhchabran — APPROVED** [review]
Incremental PRs are making things so easy to review 😊

File: dev/sg/internal/lint/lint.go (hunk 1 above)
Comment (bobheadxi) [reply dev/sg/internal/lint/lint.go:12]: Good idea 😁

**Pull request merged**